### PR TITLE
Feature/reloader logger

### PR DESCRIPTION
**Pull request by acelaya**
This provides the implementation which will make the hot code `Reloader` to use the same logger configured for access logging, as discussed in #63 

Considerations:
* I have created this as a feature, because it felt like a little bit too much to be a bugfix, but I suppose it could qualify as such. I can provide a new PR against master if this one is not correct
* I have moved the logic to pull the logger from the container to a trait which is used both by the `ReloaderFactory` and the `AccessLogFactory`. However, another approach would be to create a logger-specific factory which holds the logic and is mapped to a name that is later used to fetch the logger from the previous two factories (now that I think it, probably this second approach is cleaner. Need feedback on this).
* I have tried to add a link in the docs from the hot reloading page to the logger page, but to a specific section. I'm not sure if it will work the way I've done it, but I have been quite unsuccessful while trying to compile the docs. This is what I've tried: `[here](logging.md#container-usage)`

### CODE REVIEW

File: docs/book/v2/hot-code-reload.md [hunk 1]
```diff
@@ -40,6 +40,14 @@ return [
 ];
 ```
 
+## Logging
+
+When a file is reloaded, a *notice* line will be logged with the message
+`File changed {path}, reloading`.
+
+The logger used to log these lines is the same used for access logging,
+which is described [here](logging.md#container-usage).
```
Comment (weierophinney) [inline docs/book/v2/hot-code-reload.md]: This is definitely the correct format for the link, and our documentation renderer will correctly parse it.

File: docs/book/v2/hot-code-reload.md [hunk 2]
```diff
@@ -40,6 +40,14 @@ return [
 ];
 ```
 
+## Logging
+
+When a file is reloaded, a *notice* line will be logged with the message
+`File changed {path}, reloading`.
```
Comment (weierophinney) [inline docs/book/v2/hot-code-reload.md]: Could we maybe change this to: `Reloading due to file change: {path}`? That puts the action ("reloading") first in the message, and the variable information (path) last, which will be easier to parse using features such as grep.

File: docs/book/v2/hot-code-reload.md (hunk 1 above)
Comment (froschdesign) [reply docs/book/v2/hot-code-reload.md]: Right, the link has the correct format, but the text "here" is not very meaningful. Better would be something like: "which is described in the [Logging section](logging.md#container-usage) of this documentation"

(More informations in can be found in the W3C - QA Tips article: [Don't use "click here" as link text](https://www.w3.org/QA/Tips/noClickHere).)

Comment (acelaya) [reply docs/book/v2/hot-code-reload.md]: Agree on this

File: docs/book/v2/hot-code-reload.md (hunk 2 above)
Comment (acelaya) [reply docs/book/v2/hot-code-reload.md]: I think it's a great idea.

**Comment by acelaya** [comment]
I have just pushed proposed changes, but there are a few new considerations:

* I've called the new logger `SwooleLogger`, but just because I couldn't come with a better name.
* Now that the `AccessLogFactory` calls the new logger factory, both tests have what feels like duplicated test cases. Should I remove some of the test methods from `AccessLogFactoryTest`?

**Comment by weierophinney** [comment]
> * Should I remove some of the test methods from `AccessLogFactoryTest`?

Yes, please. Create a trait that you can compose into both tests, which will remove duplication, while ensuring they behave the same.

**Comment by acelaya** [comment]
@weierophinney I have just created the trait as requested. Let me know if that's what you had in mind.

**Comment by weierophinney** [comment]
Thanks, @acelaya!

**Comment by acelaya** [comment]
You are welcome :)

**Pull request merged**